### PR TITLE
Refactor "substr" calls in lib/private to improve code readability

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
Replacing `substr` with `str_starts_with` and `str_ends_with` in the `/lib/private` namespace to improve code readability

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: lib/private/DateTimeFormatter.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ public function formatDate($timestamp, $format = 'long', \DateTimeZone $timeZone
 	 * @return string Formatted relative date string
 	 */
 	public function formatDateRelativeDay($timestamp, $format = 'long', \DateTimeZone $timeZone = null, \OCP\IL10N $l = null) {
-		if (substr($format, -1) !== '*' && substr($format, -1) !== '*') {
+		if (!str_ends_with($format, '*') && !str_ends_with($format, '*')) {
```
Comment (fsamapoor) [inline lib/private/DateTimeFormatter.php:128]: Aren't these the same?

**Review by fsamapoor — APPROVED** [review]
Much much better. Thanks!
Other than the comment I left, looks good to me.

File: lib/private/DateTimeFormatter.php (hunk 1 above)
Comment (shdehnavi) [reply lib/private/DateTimeFormatter.php]: Thank you, it's addressed in the next commit

File: lib/private/DateTimeFormatter.php [hunk 2]
```diff
@@ -125,7 +125,7 @@ public function formatDate($timestamp, $format = 'long', \DateTimeZone $timeZone
 	 * @return string Formatted relative date string
 	 */
 	public function formatDateRelativeDay($timestamp, $format = 'long', \DateTimeZone $timeZone = null, \OCP\IL10N $l = null) {
-		if (substr($format, -1) !== '*' && substr($format, -1) !== '*') {
+		if (!str_ends_with($format, '*')) {
```
Comment (artonge) [inline lib/private/DateTimeFormatter.php:128]:
```suggestion
		if (!str_ends_with($formatDate, '^') && !str_ends_with($formatDate, '*')) {
```

I think the duplication was due to a bad copy-pasting. Having `*` or `^` at the end of the format seem on purpose and make sense if you look at the documentation of the library that we use. Same as on line 292.

https://punic.readthedocs.io/en/3.4.0/_static/api/class-Punic.Calendar.html
Screenshot from 2023-08-17 19-38-51

**Review by artonge — APPROVED** [review]

**Review by come-nc — CHANGES_REQUESTED** [review]

**Review by come-nc — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**